File: code_folder/task2.java
//создание графа на джава
static Graph createGraph() {
    Graph graph = new Graph();
    //Добавление элементов
    graph.addVertex("Bob");
    graph.addVertex("Alice");
    graph.addVertex("Mark");
    graph.addVertex("Rob");
    graph.addVertex("Maria");
    //Добавление рёбер
    graph.addEdge("Bob", "Maria");
    graph.addEdge("Bob", "Mark");
    graph.addEdge("Maria", "Alice");
    graph.addEdge("Mark", "Alice");
    graph.addEdge("Maria", "Rob");
    graph.addEdge("Mark", "Rob");

    return graph;
}

File: code_folder/task2.cpp
//пример создания графа на С++
#include <iostream>  
#include <vector>  
using namespace std;  
class Graph {  
// Матрица инцидентности для хранения рёбер графа  
vector<vector<int> > adj_matrix;  
public:  
//Создание графа с n вершинами  
Graph(int n) { adj_matrix = vector<vector<int> >(n, vector<int>(n, 0)); }  
//Добавление ребра между вершинами u и v  
void add_edge(int u, int v) {  
// Установить ребро от u к v  
adj_matrix[u][v] = 1;  
// Установить ребро от v к u (для неориентированного графа)  
adj_matrix[v][u] = 1;  
}  
// Функция для печати представления графа с использованием матрицы инцидентности  
void print() {  
// Итерироваться по каждой вершине  
for (auto i : adj_matrix) {  
// Вывести вершину  
cout << i.first << " -> ";  
// Итерироваться по связанным вершинам  
for (auto j : i.second) {  
// Вывести связанную вершину  
cout << j << " ";  
}  
cout << endl;  
}  
}  
};  
int main() {  
// Создать объект графа  
Graph g;  
// Добавление ребер в граф  
g.add_edge(1, 0);  
g.add_edge(2, 0);  
g.add_edge(1, 2);  
//вывод графа
g.print();  
return 0;  
}
